ckanext/xloader/plugin.py

```diff
@@ -70,17 +71,33 @@ def configure(self, config_):
     # IDomainObjectModification
 
     def notify(self, entity, operation):
-        # type: (ckan.model.Package|ckan.model.Resource, DomainObjectOperation) -> None
+        # type: (Package|Resource, DomainObjectOperation) -> None
         """
         Runs before_commit to database for Packages and Resources.
         We only want to check for changed Resources for this.
         We want to check if values have changed, namely the url.
         See: ckan/model/modification.py.DomainObjectModificationExtension
         """
         if operation != DomainObjectOperation.changed \
-                or not isinstance(entity, Resource) \
-                or not getattr(entity, 'url_changed', False):
+                or not isinstance(entity, Resource):
             return
+
+        # If the resource requires validation, stop here if validation
+        # has not been performed or did not succeed. The Validation
+        # extension will call resource_patch and this method should
+        # be called again. However, url_changed will not be in the entity
+        # once Validation does the patch.
+        if utils.is_validation_plugin_loaded() and \
+                toolkit.asbool(toolkit.config.get('ckanext.xloader.requires_validation')):
+
+            if entity.__dict__.get('extras', {}).get('validation_status', None) != 'success':
```

Comment:
This won't work for us, because validation status will be blank for resources without a schema (which is most of them on our site).